odl/space/cu_ntuples.py

```diff
@@ -847,36 +976,271 @@ def _norm_default(x):
     return x.data.norm()
 
 
+def _pnorm_default(x, p):
+    if p == float('inf'):
+        raise NotImplementedError('inf-norm not implemented.')
+    elif int(p) != p:
+        raise NotImplementedError('non-integer powers not implemented')
+    # TODO: optimized version in C++ code?
```

Comment:
I'd like to know how it works. I actually had a quick look into the C++ code, but there was no obvious place to steal code from, so I decided not to invest more time searching around. Are there any `thrust` transform functions one can use or do we need a custom functor as last time (don't remember which function it was for)?